### PR TITLE
[iPhoneX] Converts the bottom nav to work on iPhone X

### PR DESCRIPTION
**Pull request by orta**
- [x] The bottom nav - https://github.com/artsy/eigen/commit/09a7cd87577c6011e80c22e656147242dccd76b7

screen shot 2018-01-17 at 2 18 27 pm

#trivial

### CODE REVIEW

File: Artsy/Categories/Apple/UIDevice-Hardware.h [hunk 1]
```diff
@@ -1,87 +1,5 @@
...
-
-typedef NS_ENUM(NSUInteger, UIDeviceFamily) {
-    UIDeviceFamilyiPhone,
-    UIDeviceFamilyiPod,
-    UIDeviceFamilyiPad,
-    UIDeviceFamilyAppleTV,
-    UIDeviceFamilyUnknown,
-};
-
-
```
Comment (orta) [inline Artsy/Categories/Apple/UIDevice-Hardware.h]: unused

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Developer Specific file shouldn’t be changed</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m [hunk 2]
```diff
@@ -390,17 +391,26 @@ - (void)setNotificationCount:(NSUInteger)number forControllerAtIndex:(ARTopTabCo
...
+    if (@available(iOS 11.0, *)) {
+        return self.view.safeAreaInsets.bottom * -1;
+    } else {
+        return 0;
+    }
+}
+
 #pragma mark - ARMenuAwareViewController
 
 - (void)hideToolbar:(BOOL)hideToolbar animated:(BOOL)animated
```
Comment (orta) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m]: Showing the first VC (Home)  would trigger `hideToolbar` with `NO` - which triggers setting  the bottom constraint before we present the screen for the first time. Can't set this stuff up at `viewDidLoad` before views are somewhat on the screen.

**Comment by orta** [comment]
This is red from Danger - the rule is correct, most people edit this file by accident. 

However, my use case is legit from the version bump, so I'm merging with red.

**Pull request merged**